api/src/main/java/org/pmiops/workbench/reporting/ReportingServiceImpl.java

```diff
@@ -1,24 +1,52 @@
 package org.pmiops.workbench.reporting;
 
+import javax.inject.Provider;
+import org.pmiops.workbench.config.WorkbenchConfig;
 import org.pmiops.workbench.model.ReportingSnapshot;
+import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.stereotype.Service;
 
 @Service
 public class ReportingServiceImpl implements ReportingService {
 
-  private final ReportingUploadService reportingUploadService;
+  private static enum UploadMethod {
+    DML,
+    STREAMING;
+  }
+
   private final ReportingSnapshotService reportingSnapshotService;
+  private final ReportingUploadService reportingUploadServiceDmlImpl;
+  private final ReportingUploadService reportingUploadServiceStreamingImpl;
+  private final Provider<WorkbenchConfig> workbenchConfigProvider;
 
   public ReportingServiceImpl(
-      ReportingUploadService reportingUploadService,
-      ReportingSnapshotService reportingSnapshotService) {
-    this.reportingUploadService = reportingUploadService;
+      @Qualifier("REPORTING_UPLOAD_SERVICE_DML_IMPL")
+          ReportingUploadService reportingUploadServiceDmlImpl,
+      @Qualifier("REPORTING_UPLOAD_SERVICE_STREAMING_IMPL")
+          ReportingUploadService reportingUploadServiceStreamingImpl,
+      ReportingSnapshotService reportingSnapshotService,
+      Provider<WorkbenchConfig> workbenchConfigProvider) {
+    this.reportingUploadServiceDmlImpl = reportingUploadServiceDmlImpl;
+    this.reportingUploadServiceStreamingImpl = reportingUploadServiceStreamingImpl;
+    this.workbenchConfigProvider = workbenchConfigProvider;
     this.reportingSnapshotService = reportingSnapshotService;
   }
 
   @Override
   public ReportingJobResult takeAndUploadSnapshot() {
     final ReportingSnapshot snapshot = reportingSnapshotService.takeSnapshot();
-    return reportingUploadService.uploadSnapshot(snapshot);
+    return getConfiguredUploadService().uploadSnapshot(snapshot);
+  }
+
+  private ReportingUploadService getConfiguredUploadService() {
+    final String uploadMethod = workbenchConfigProvider.get().reporting.uploadMethod;
+    if (UploadMethod.DML.name().equals(uploadMethod)) {
```

Comment:
Sadly we can't do this in the DI, because the config isn't available at bootstrap time. So I'm always injecting dead code unless there's a runtime change to it.

Though perhaps I could manual-wire it instead of using ctor injection. Not really worth that though.